src/pmhn/_trees/_backend.py

```diff
@@ -2,9 +2,16 @@
 
 
 import numpy as np
+from pmhn._trees._interfaces import Tree
+from pmhn._trees._tree_utils import create_all_subtrees, bfs_compare
+from anytree import Node, LevelOrderGroupIter
 
 
-from pmhn._trees._interfaces import Tree
+class LoglikelihoodSingleTree:
+    def __init__(self, tree: Node):
+        self._subtrees_dict = create_all_subtrees(tree)
+
+    _subtrees_dict: dict[Node, int]
```

Comment:
I'd remove line 14 as it [creates a class attribute, rather than an instance attribute](https://builtin.com/software-engineering-perspectives/python-attributes) and move the type annotation (great you have added it!) to line 12.